pkg/aggregator/mocksender/asserts.go

```diff
@@ -46,6 +47,54 @@ func (m *MockSender) AssertMetricNotTaggedWith(t *testing.T, method string, metr
 	return m.Mock.AssertCalled(t, method, metric, mock.AnythingOfType("float64"), mock.AnythingOfType("string"), AssertTagsNotContains(tags))
 }
 
+// Compare an Event on specifics values
+func eventHaveEqualValues(expectedEvent, actualEvent metrics.Event) bool {
+	if assert.ObjectsAreEqualValues(expectedEvent.AggregationKey, actualEvent.AggregationKey) &&
+		assert.ObjectsAreEqualValues(expectedEvent.Priority, actualEvent.Priority) &&
+		assert.ObjectsAreEqualValues(expectedEvent.SourceTypeName, actualEvent.SourceTypeName) &&
+		assert.ObjectsAreEqualValues(expectedEvent.EventType, actualEvent.EventType) &&
+		expectedInActual(expectedEvent.Tags, actualEvent.Tags) {
+		return true
+	}
+	return false
+}
+
+// Iter on all mock.Calls to find any events who's matching the expectedEvent
+// The check on the Event.Ts is weighted with the parameter allowedDelta
+func (m *MockSender) matchEvent(expectedEvent metrics.Event, allowedDelta time.Duration) bool {
```

Comment:
Could you please move it to a MatchedBy so it's usable in custom `mock.AssertCalled` invocations? Should be something like:

```go
func MatchEventLike(expected metrics.Event, allowedDelta time.Duration) interface{} {
	return mock.MatchedBy(func(actual metrics.Event) bool {
		expectedTime = time.Unix(expected.Ts, 0)
		actualTime = time.Unix(actual.Ts, 0)
		dt = expectedTime.Sub(actualTime)
		if dt < -allowedDelta || dt > allowedDelta {
			return false
		}
		return eventHaveEqualValues(expected, actual)
	})
}
```